py_zipkin/stack.py

```diff
@@ -23,6 +23,23 @@ def get(self):
         if self._storage:
             return self._storage[-1]
 
+    def __iter__(self):
+        self.index = 0
+        return self
+
+    def __next__(self):  # pragma: no cover
+        return self.next()
+
+    def next(self):
+        if len(self._storage) <= self.index:
+            raise StopIteration
```

Comment:
I think it is more canonical to have generators consume the items as they iterate. The advantage of having the storage consume the items is that you won't have to call `clear` at the end.

That of course would be O(N) for the default thread local attrs, since its a list (array). But if we change the thread local attrs to use a `deque`, we could do this in constant time (we'd want to make sure to communicate this access pattern in the docstrings so that the implementors of span storage can be sure to use a deque/linked-list).

Another alternative would be to have a method to return an iterator over the items and to not have the storage itself be iterable.

Reply:
We can't have the generator consume the items since we have to iterate twice. Once for the transport_handler and once for the firehose_handler.

Reply:
Oh right. My concern is that by manually clearing the storage, we might add a new code path in the future and forget to clear the span store. Is there another way that we can ensure that it gets cleared?

Reply:
I integrated some of these comments. I now have `SpanStore` which is basically a deque. 
Deques are iterable, so I don't have to explicitly write the iterator logic. They also already have a `clear` method to wipe their content.

I haven't found a way to avoid calling `span_storage.clear()` unless we want to do a deepcopy of it before calling `emit_spans`.